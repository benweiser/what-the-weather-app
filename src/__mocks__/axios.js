export default {
  get: jest.fn(() => Promise.resolve({ data: {} })),
  request: jest.fn(() => Promise.resolve({ data: {} }))
};

const localStorageMock = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  clear: jest.fn()
};
global.localStorage = localStorageMock;
global.sessionStorage = localStorageMock;
