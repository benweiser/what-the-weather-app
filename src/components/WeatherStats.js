import React from "react";
import Typography from "@material-ui/core/Typography";

import CurrentWeatherIcon from "./CurrentWeatherIcon";
import GoogleMap from "./GoogleMap";

const WeatherStats = ({ data }) => {
  if (!data) {
    return null;
  }
  const {
    coord: { lat, lon } = {},
    main: { temp, temp_min, temp_max, humidity } = {},
    name,
    sys: { sunset, sunrise } = {},
    weather,
    wind: { deg, speed } = {}
  } = data;

  return (
    <div data-testid="WeatherStats">
      <Typography variant="headline" component="h3" gutterBottom>
        City: {name}
      </Typography>
      <CurrentWeatherIcon currentConditions={weather[0].id} />
      <p>
        Latitude: {lat}, Longitude: {lon}
      </p>
      <p>Current Temperature: {temp}</p>
      <p>Today's Low: {temp_min}</p>
      <p>Today's High: {temp_max}</p>
      <p>Current Humidity: {humidity}</p>
      <div>
        Current Conditions:{" "}
        {weather.map(condition => (
          <div key={`${condition.main}`}>{condition.main}</div>
        ))}
      </div>
      <p>Sunset Time: {sunset}</p>
      <p>Sunrise Time: {sunrise}</p>
      <p>Wind Speed: {speed}</p>
      <p>Wind Degree: {deg}</p>

      <GoogleMap
        center={{
          lat,
          lng: lon
        }}
      />
    </div>
  );
};

export default WeatherStats;
