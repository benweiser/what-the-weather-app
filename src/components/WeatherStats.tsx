import React, { memo, Suspense } from 'react';
import Typography from '@material-ui/core/Typography';

import CurrentWeatherIcon from './CurrentWeatherIcon';
import { CurrentWeatherStats } from '../services/api/WeatherService';
import GoogleMap from './GoogleMap';

export interface WeatherStatsProps {
  data: CurrentWeatherStats;
}

const WeatherStats = ({ data }: WeatherStatsProps) => {
  if (!data) {
    return null;
  }

  const {
    currentConditions,
    lat,
    locationName,
    lon,
    sunsetTime,
    sunriseTime,
    temp,
    tempMax,
    tempMin,
    windSpeed
  } = data;

  return (
    <div data-testid="WeatherStats">
      {locationName && (
        <Typography variant="headline" component="h3" gutterBottom>
          City: {locationName}
        </Typography>
      )}
      {currentConditions && (
        <CurrentWeatherIcon currentConditions={currentConditions[0].id} />
      )}

      {lat && lon && (
        <p>
          Latitude: {lat}, Longitude: {lon}
        </p>
      )}

      {temp && <p>Current Temperature: {temp}</p>}
      {tempMin && <p>Today&#39;s Low: {tempMin}</p>}
      {tempMax && <p>Today&#39;s High: {tempMax}</p>}
      {currentConditions && (
        <div>
          Current Conditions:{' '}
          {currentConditions.map(condition => (
            <div key={`${condition.main}`}>{condition.main}</div>
          ))}
        </div>
      )}
      {sunsetTime && <p>Sunset Time: {sunsetTime}</p>}
      {sunriseTime && <p>Sunrise Time: {sunriseTime}</p>}
      {windSpeed && <p>Wind Speed: {windSpeed}</p>}

      {lat && lon && (
        <GoogleMap
          center={{
            lat,
            lng: lon
          }}
        />
      )}
    </div>
  );
};

export default WeatherStats;
