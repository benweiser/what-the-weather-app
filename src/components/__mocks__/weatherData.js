export const mockWeatherData = {
  name: 'Test',
  coord: {
    lat: 23,
    lon: 32
  },
  weather: [
    {
      id: 500,
      description: 'Test description'
    }
  ],
  main: {
    humidity: 40,
    temp: 45,
    temp_min: 23,
    temp_max: 79
  },
  sys: {
    sunrise: '5:00',
    sunset: '8:00'
  },
  wind: {
    speed: 43,
    deg: 43
  }
};
