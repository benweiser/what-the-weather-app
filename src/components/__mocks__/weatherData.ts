import { CurrentWeatherStats } from './../../services/api/WeatherService';

export const mockRawWeatherData = {
  name: 'Test',
  coord: {
    lat: 23,
    lon: 32
  },
  weather: [
    {
      id: 500,
      icon: 'xyz',
      description: 'Test description',
      main: 'Test'
    }
  ],
  main: {
    humidity: 40,
    temp: 45,
    temp_min: 23,
    temp_max: 79
  },
  sys: {
    sunrise: 1571500519351,
    sunset: 1571500619351
  },
  wind: {
    speed: 43,
    deg: 43
  }
};

export const mockCurrentWeatherStats: CurrentWeatherStats = {
  currentConditions: [
    {
      description: 'Test description',
      icon: 'xyz',
      id: 500,
      main: 'Test'
    }
  ],
  lat: 23,
  locationName: 'Test',
  lon: 32,
  sunriseTime: 1571500519351,
  sunsetTime: 1571500619351,
  temp: 45,
  tempMax: 79,
  tempMin: 23,
  windSpeed: 43
};
