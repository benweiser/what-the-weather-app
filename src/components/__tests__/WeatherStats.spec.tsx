import { render, waitForElement } from 'react-testing-library';
import React, { Suspense } from 'react';
import WeatherStats from '../WeatherStats';
import { mockCurrentWeatherStats } from '../__mocks__/weatherData';

describe('Weather Stats', () => {
  it('renders', async () => {
    const { container } = render(
      <Suspense fallback="...">
        <WeatherStats data={mockCurrentWeatherStats} />
      </Suspense>
    );
    const lazyElement = await waitForElement(() => container);
    expect(lazyElement).toMatchSnapshot();
  });
});
