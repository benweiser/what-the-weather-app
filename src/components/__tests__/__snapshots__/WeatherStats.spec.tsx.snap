// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Stats renders 1`] = `
.c0 {
  font-size: 64px;
}

<div>
  <div
    data-testid="WeatherStats"
  >
    <h3
      class="MuiTypography-root-1 MuiTypography-headline-6 MuiTypography-gutterBottom-18"
    >
      City: 
      Test
    </h3>
    <span
      class="wi wi-day-rain c0"
      data-testid="CurrentWeatherIcon"
    />
    <p>
      Latitude: 
      23
      , Longitude: 
      32
    </p>
    <p>
      Current Temperature: 
      45
    </p>
    <p>
      Today's Low: 
      23
    </p>
    <p>
      Today's High: 
      79
    </p>
    <div>
      Current Conditions:
       
      <div>
        Test
      </div>
    </div>
    <p>
      Sunset Time: 
      8:00
    </p>
    <p>
      Sunrise Time: 
      5:00
    </p>
    <p>
      Wind Speed: 
      43
    </p>
    <div
      style="height: 400px; width: 100%;"
    >
      <div
        style="width: 100%; height: 100%; margin: 0px; padding: 0px; position: relative;"
      >
        <div
          style="width: 100%; height: 100%; left: 0px; top: 0px; margin: 0px; padding: 0px; position: absolute;"
        />
        <div
          style="width: 50%; height: 50%; left: 50%; top: 50%; margin: 0px; padding: 0px; position: absolute;"
        >
          <div
            style="width: 100%; height: 100%; left: 0px; top: 0px; margin: 0px; padding: 0px; position: absolute;"
          >
            <div
              style="width: 0px; height: 0px; left: 0px; top: 0px; background-color: transparent; position: absolute;"
            >
              <div
                style="color: white; font-size: 21px;"
              >
                <i
                  aria-hidden="true"
                  class="fa fa-map-marker"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
